### PR TITLE
Log collector takes resource group as input

### PR DESCRIPTION
**Pull request by Bhuvaneswari-Santharam**

### CODE REVIEW

File: diagnosis/getkuberneteslogs.sh [hunk 1]
```diff
@@ -191,6 +165,27 @@ if [ $? -ne 0 ]; then
     exit 1
 fi
 
+#checks if azure-cli is installed   
+requirements
```
Comment (jadarsie) [inline diagnosis/getkuberneteslogs.sh]: `checkRequirements` may be a slightly more explanatory name

File: diagnosis/getkuberneteslogs.sh [hunk 2]
```diff
@@ -191,6 +165,27 @@ if [ $? -ne 0 ]; then
...
+
+#workaround for SSL interception
+export AZURE_CLI_DISABLE_CONNECTION_VERIFICATION=1 
+export ADAL_PYTHON_SSL_NO_VERIFY=1
+
+#Validate resource-group
+location=$(az group show -n $RESOURCE_GROUP --query location)
+if [ $? -ne 0 ]; then
+        echo "[$(date +%Y%m%d%H%M%S)][ERR] Specified Resource group not found."
```
Comment (jadarsie) [inline diagnosis/getkuberneteslogs.sh]: Please run VS Code extension Bash Beautify to fix format

File: diagnosis/getkuberneteslogs.sh [hunk 3]
```diff
@@ -191,6 +165,27 @@ if [ $? -ne 0 ]; then
     exit 1
 fi
 
+#checks if azure-cli is installed   
+requirements
+
+#workaround for SSL interception
+export AZURE_CLI_DISABLE_CONNECTION_VERIFICATION=1 
+export ADAL_PYTHON_SSL_NO_VERIFY=1
```
Comment (jadarsie) [inline diagnosis/getkuberneteslogs.sh]: We may want to rollback this change after the log collector is done. See the `trap` command.

File: diagnosis/getkuberneteslogs.sh [hunk 4]
```diff
@@ -191,6 +165,27 @@ if [ $? -ne 0 ]; then
...
+
+#Validate resource-group
+location=$(az group show -n $RESOURCE_GROUP --query location)
+if [ $? -ne 0 ]; then
+        echo "[$(date +%Y%m%d%H%M%S)][ERR] Specified Resource group not found."
+        exit 1
+fi
+
+#Get the master nodes from the resource group
+master_nodes=$(az resource list -g $RESOURCE_GROUP --resource-type "Microsoft.Compute/virtualMachines" --query "[?tags.poolName=='master'].{Name:name}" --output table)
```
Comment (jadarsie) [inline diagnosis/getkuberneteslogs.sh]: I think using `az vm list` is more standard. E.x.:
`az vm list -g $rg --query "[?tags.poolName=='master'].{Name:name}" --output tsv`

Output type `tsv` will simplify looping over the list

File: diagnosis/getkuberneteslogs.sh [hunk 5]
```diff
@@ -191,6 +165,27 @@ if [ $? -ne 0 ]; then
...
+#checks if azure-cli is installed   
+requirements
+
+#workaround for SSL interception
+export AZURE_CLI_DISABLE_CONNECTION_VERIFICATION=1 
+export ADAL_PYTHON_SSL_NO_VERIFY=1
+
+#Validate resource-group
+location=$(az group show -n $RESOURCE_GROUP --query location)
+if [ $? -ne 0 ]; then
```
Comment (jadarsie) [inline diagnosis/getkuberneteslogs.sh]: az commands always return exit code 0. Check for `$?` is not a good test in this case.

File: diagnosis/getkuberneteslogs.sh [hunk 6]
```diff
@@ -138,28 +128,14 @@ else
     || { echo "The identity file $IDENTITYFILE is not a RSA Private Key file."; echo "A RSA private key file starts with '-----BEGIN [RSA|OPENSSH] PRIVATE KEY-----''"; exit 1; }
 fi
 
-if [ -z "$SPN_CLIENT_ID" -a -z "$SPN_CLIENT_SECRET" ] && [ -n "$UPLOAD_LOGS" ]
+if [ -z "$RESOURCE_GROUP" ]
 then
     echo ""
-    echo "[ERR] Service Principal details should be provided if logs are stored in a storage account"
+    echo "[ERR] Resource group should be provided"
```
Comment (jadarsie) [inline diagnosis/getkuberneteslogs.sh]: Let's use the same error message for required parameters, see `--user`

File: diagnosis/getkuberneteslogs.sh [hunk 7]
```diff
@@ -191,6 +165,27 @@ if [ $? -ne 0 ]; then
...
+fi
+
+#Get the master nodes from the resource group
+master_nodes=$(az resource list -g $RESOURCE_GROUP --resource-type "Microsoft.Compute/virtualMachines" --query "[?tags.poolName=='master'].{Name:name}" --output table)
+if [ $? -ne 0 ]; then
+        echo "[$(date +%Y%m%d%H%M%S)][ERR] Kubernetes master nodes not found in the resource group."
+        exit 1
+fi
+
 if [ -n "$MASTER_HOST" ]
```
Comment (jadarsie) [inline diagnosis/getkuberneteslogs.sh]: We can remove flag `MASTER_HOST` once we obtain the master nodes using azure-cli

File: diagnosis/getkuberneteslogs.sh (hunk 5 above)
Comment (Bhuvaneswari-Santharam) [reply diagnosis/getkuberneteslogs.sh]: Nope, if the resource group is not found the exit code is 3.

File: diagnosis/getkuberneteslogs.sh (hunk 7 above)
Comment (Bhuvaneswari-Santharam) [reply diagnosis/getkuberneteslogs.sh]: Will do that in next iteration

Comment (Bhuvaneswari-Santharam) [reply diagnosis/getkuberneteslogs.sh]: I will get the master ip and then remove this flag in next iteration

File: diagnosis/getkuberneteslogs.sh [hunk 8]
```diff
@@ -1,6 +1,16 @@
 #!/bin/bash
+restoreAzureCLIVariables()
+{
+    EXIT_CODE=$?
+    #restoring Azure CLI values
+    export AZURE_CLI_DISABLE_CONNECTION_VERIFICATION=0
```
Comment (jadarsie) [inline diagnosis/getkuberneteslogs.sh]: We need to set it back to the original value

File: diagnosis/getkuberneteslogs.sh (hunk 5 above)
Comment (jadarsie) [reply diagnosis/getkuberneteslogs.sh]: you are right, sorry for that

File: diagnosis/getkuberneteslogs.sh (hunk 6 above)
Comment (jadarsie) [reply diagnosis/getkuberneteslogs.sh]: I will make this change myself

File: diagnosis/getkuberneteslogs.sh (hunk 4 above)
Comment (jadarsie) [reply diagnosis/getkuberneteslogs.sh]: I can make this change myself

**Pull request merged**